index.js

```diff
@@ -272,8 +272,8 @@ class Ky {
 		this.request = new globals.Request(this._input, this._options);
 
 		if (this._options.searchParams) {
-			const url = new URL(this.request.url);
-			url.search = new URLSearchParams(this._options.searchParams);
+			const searchParams = '?' + new URLSearchParams(this._options.searchParams).toString();
+			const url = this.request.url.replace(/(?:\?.*?)?(?=#|$)/, searchParams);
```

Comment:
```suggestion
			const url = this.request.url.replace(/(?:\?.*?)(?=#|$)/, searchParams);
```

?

Because `.*` already means 0 or more matches.

Reply:
That wouldn't work in the case where the URL is missing a query string and we want to add one.

The non-capturing group `(?:\?.*?)` matches an existing query string. But we don't know if the URL has one, so the group is optional.

I added tests to cover this, which should fail if you make the suggested change. The relevant assertion is:

```js
 t.is(await ky('/unicorn', {searchParams: {foo: 'bar'}}).text(), '/unicorn?foo=bar');
```

Reply:
What about defining the regex as a constant, so V8 doesn't have to create it every time?